stl/inc/ranges

```diff
@@ -411,31 +421,45 @@ namespace ranges {
     };
 
     // clang-format off
-    template <_Copy_constructible_object _Ty>
-        requires (copyable<_Ty>
-            || (is_nothrow_copy_constructible_v<_Ty>
-                && (movable<_Ty> || is_nothrow_move_constructible_v<_Ty>)))
-    class _Copyable_box<_Ty> { // provide the same API more efficiently when we can avoid the disengaged state
+    template <class _Ty>
+    concept _Use_simple_movable_box_wrapper =
+        // [range.move.wrap]
+        // 1. If copy_constructible<T> is true, movable-box<T> should store only a T if either T models copyable, or
+        // is_nothrow_move_constructible_v<T> && is_nothrow_copy_constructible_v<T> is true.
+        (copy_constructible<_Ty> && (

```

Comment:
This condition is being phrased as `(A && (B)) || (!A && (C))`. Suggestion: `A ? (B) : (C)` would be simpler and match the Standardese more closely.

(I wrote a comment about this in my initial review, but it didn't appear - this happens occasionally and I haven't figured out whether it's a GitHub bug or if I'm just forgetting to publish the comment and nothing warns me. Sorry for not including this in the initial batch of comments!)

Reply:
I talked to Casey about this originally, and we decided that `?:` was weird; I'm okay switching to it though, given that this is very weird anyways.